WordPress/Classes/ViewRelated/Me/Change Username/View Model/ChangeUsernameViewModel.swift

```diff
@@ -132,6 +143,13 @@ private extension ChangeUsernameViewModel {
             static let validateUsername = NSLocalizedString("There was an error validating the username", comment: "Text displayed when there is a failure validating the username.")
             static let saveUsername = NSLocalizedString("There was an error saving the username", comment: "Text displayed when there is a failure saving the username.")
             static let emptyValue = NSLocalizedString("Username must be at least 4 characters.", comment: "Error displayed if the username input is an empty string")
+            static let alphanumeric = NSLocalizedString("Username can only contain lowercase letters (a-z) and numbers.", comment: "Error displayed if the username doesn't contain numbers or digits")
         }
     }
 }
+
+private extension String {
+    var isAlphanumeric: Bool {
+        return range(of: "[^a-zA-Z0-9]", options: .regularExpression) == nil
```

Comment:
Are usernames allowed accented characters at all? (I don't know the answer to this) I suspect not, but if they are we may want to do something like

`rangeOfCharacter(from: CharacterSet.alphanumerics.inverted) == nil`

Reply:
>rangeOfCharacter(from: CharacterSet.alphanumerics.inverted) == nil

This was my first attempt. Using characters sets to check the string, but it includes the accents. The regex then was the simple solution to get only alphanumerics characters ignoring diacritics.